tests/utils/streams/mixins.py

```diff
@@ -0,0 +1,159 @@
+                consumer.seek({Partition(topic, 0): 0})
+
+            # TODO: It'd be much nicer if ``subscribe`` returned a future that we could
+            # use to wait for assignment, but we'd need to be very careful to avoid
+            # edge cases here. It's probably not worth the complexity for now.
+            consumer.subscribe(
+                [topic], on_assign=assignment_callback, on_revoke=revocation_callback
+            )
+
+            message = consumer.poll(10.0)  # XXX: getting the subcription is slow
+            assert isinstance(message, Message)
+            assert message.partition == Partition(topic, 0)
+            assert message.offset == 1
+            assert message.payload == 1
+
+            assert consumer.tell() == {Partition(topic, 0): 2}
+            assert getattr(assignment_callback, "called", False)
+
+            consumer.seek({Partition(topic, 0): 0})
+            assert consumer.tell() == {Partition(topic, 0): 0}
+
+            with pytest.raises(ConsumerError):
+                consumer.seek({Partition(topic, 1): 0})
+
+            consumer.pause([Partition(topic, 0)])
+
+            consumer.resume([Partition(topic, 0)])
+
+            message = consumer.poll(1.0)
+            assert isinstance(message, Message)
+            assert message.partition == Partition(topic, 0)
+            assert message.offset == 0
+            assert message.payload == 0
+
+            assert consumer.commit_offsets() == {}
+
+            consumer.stage_offsets({message.partition: message.get_next_offset()})
+
+            with pytest.raises(ConsumerError):
+                consumer.stage_offsets({Partition(Topic("invalid"), 0): 0})
+
+            assert consumer.commit_offsets() == {
+                Partition(topic, 0): message.get_next_offset()
+            }
+
+            consumer.unsubscribe()
+
+            assert consumer.poll(1.0) is None
+
+            assert consumer.tell() == {}
+
+            with pytest.raises(ConsumerError):
+                consumer.seek({Partition(topic, 0): 0})
+
+            consumer.close()
+
+            with pytest.raises(RuntimeError):
+                consumer.subscribe([topic])
+
+            with pytest.raises(RuntimeError):
+                consumer.unsubscribe()
+
+            with pytest.raises(RuntimeError):
+                consumer.poll()
+
+            with pytest.raises(RuntimeError):
+                consumer.tell()
+
+            with pytest.raises(RuntimeError):
+                consumer.seek({Partition(topic, 0): 0})
+
+            with pytest.raises(RuntimeError):
+                consumer.pause([Partition(topic, 0)])
+
+            with pytest.raises(RuntimeError):
+                consumer.resume([Partition(topic, 0)])
+
+            with pytest.raises(RuntimeError):
+                consumer.stage_offsets({})
+
+            with pytest.raises(RuntimeError):
+                consumer.commit_offsets()
+
+            consumer.close()
```

Comment:
Didn't you close it before ? Or is this to test that you can close twice without throwing ?

Reply:
Yeah, this is to test that calling close doesn't raise if the instance is already closed. I'll add a comment here to clarify.

Reply:
Added some clarifications in 75f1205f3a48fe46b5a7cb7bde22ff0810411483.